### PR TITLE
Format on type

### PR DESCRIPTION
**Pull request by jrieken**

### CODE REVIEW

**Comment by nosami** [comment]
Where are the tests?? :)

**Comment by jrieken** [comment]
:laughing:

File: src/OmniSharp/Api/Formatting/OmnisharpController.FormatOnType.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+using Microsoft.AspNet.Mvc;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Text;
+using OmniSharp.Models;
+
+namespace OmniSharp
+{
+
```
Comment (davidfowl) [inline src/OmniSharp/Api/Formatting/OmnisharpController.FormatOnType.cs]: Remove extra space

File: src/OmniSharp/Api/Formatting/OmnisharpController.FormatOnType.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    public partial class OmnisharpController
+    {
+        [HttpPost("formatOnType")]
+        public async Task<FormatRangeResponse> FormatOnType([FromBody]Request request)
+        {
+            _workspace.EnsureBufferUpdated(request);
+            
+            var edits = new List<TextEdit>();
+            var ret = new FormatRangeResponse() { Edits = edits };
```
Comment (davidfowl) [inline src/OmniSharp/Api/Formatting/OmnisharpController.FormatOnType.cs]: response

File: src/OmniSharp/Models/FormatRangeResponse.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        }
+
+        public string NewText { get; set; }
+        public int StartLine { get; set; }
+        public int StartColumn { get; set; }
+        public int EndLine { get; set; }
+        public int EndColumn { get; set; }
+    }
+
+    public class FormatRangeResponse
```
Comment (davidfowl) [inline src/OmniSharp/Models/FormatRangeResponse.cs]: Omg 2 classes in a single file

File: tests/OmniSharp.Tests/FormattingFacts.cs [hunk 4]
```diff
@@ -0,0 +1,112 @@
+using System;
+using Xunit;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace OmniSharp.Tests {
```
Comment (nosami) [inline tests/OmniSharp.Tests/FormattingFacts.cs]: These tests do not test your code. And what is up with the spacing??

**Comment by Yantrio** [comment]
Any chance we could get the tests to look at what responses are expected? we don't have any docs, and it would be nice to see if it actually works and how it actually works.

File: tests/OmniSharp.Tests/FormattingFacts.cs [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+	public void M ()
+	{
+		for(var i = 0; i < 8;$)
+	}
+}");
+		}
+		
+		private void AssertFormatTargetKind(SyntaxKind kind, string source) 
+		{
+			var tuple = GetTreeAndOffset(source);
```
Comment (davidfowl) [inline tests/OmniSharp.Tests/FormattingFacts.cs]: omg tabs!

File: src/OmniSharp/Workers/Format/TextChange.cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+namespace OmniSharp
+{
+    public class TextChange
+    {
+        public string NewText { get; set; }
+        public int StartLine { get; set; }
+        public int StartColumn { get; set; }
+        public int EndLine { get; set; }
+        public int EndColumn { get; set; }
+	}
```
Comment (davidfowl) [inline src/OmniSharp/Workers/Format/TextChange.cs]: wtf formatting

**Pull request merged**